stub.go

```diff
@@ -264,6 +268,15 @@ func (s *stubLndClient) generateHtlcs(key route.Vertex, peer *stubPeer,
 		chanIds = append(chanIds, chanId)
 	}
 
+	// If the peer has a closed channel, include it as one of the possible
+	// incoming channels. While this is improbable that we'd intercept a
+	// HTLC on a closed channel in practice (it's more likely that we get
+	// a resolution from a channel that has since closed), this does allow
+	// us to test the closed channel path.
+	if peer.closedChannelId != 0 {
+		chanIds = append(chanIds, peer.closedChannelId)
```

Comment:
Adding it here, but `chanCount` isn't incremented. Maybe the random below doesn't pick it up then?

In general, I wonder if it isn't better to just add a `closed` boolean to `stubChannel`?

Reply:
Much nicer, updated stub!